### PR TITLE
Add monitoring settings to reference configuration

### PR DESCRIPTION
**Pull request by tsg**
Note that monitoring is disabled by default in Beats.

Related to #6235.

### CODE REVIEW

File: libbeat/_meta/config.reference.yml [hunk 1]
```diff
@@ -784,6 +784,93 @@ logging.files:
...
+
+  # The maximum number of events to bulk in a single Elasticsearch bulk API index request.
+  # The default is 50.
+  #bulk_max_size: 50
+
+  # Configure http request timeout before failing an request to Elasticsearch.
+  #timeout: 90
+
+  # Use SSL settings for HTTPS. Default is true.
+  #ssl.enabled: true
```
Comment (urso) [inline libbeat/_meta/config.reference.yml]: Comment is somewhat misleading. This default config has no SSL settings enabled. The `ssl.enabled` setting is only used to 'ignore' the ssl settings if `ssl.enabled: false`. Plus, if "https" scheme is used system certificates will be used by the stdlib.

File: libbeat/_meta/config.yml [hunk 2]
```diff
@@ -91,3 +91,18 @@ output.elasticsearch:
...
+# reporting is disabled by default.
+
+# Set to true to enable the monitoring reporter.
+#xpack.monitoring.enabled: false
+
+# Uncomment to send the metrics to Elasticsearch. Most settings from the
+# elasticsearch output are accepted here as well. Any setting that is not set is
+# automatically inherited from the output configuration, so if you want to use
+# the same cluster for monitoring data, you can simply uncomment the following
+# line.
```
Comment (urso) [inline libbeat/_meta/config.yml]: x-pack always requires to pass data through production cluster. That is hosts must match the elasticsearch output.

>  so if you want to use the same cluster for monitoring data, you can simply uncomment the following
 line.

Note: elasticsearch report configs are only inherrited from output.elasticsearch config. If logstash is configured as output, the host+username+password for ES must be configured here.

File: libbeat/_meta/config.reference.yml (hunk 1 above)
Comment (tsg) [reply libbeat/_meta/config.reference.yml]: I just copied this from the ES output SSL settings :)

File: libbeat/_meta/config.yml (hunk 2 above)
Comment (tsg) [reply libbeat/_meta/config.yml]: Ah, I forgot about that. Let's see if I can clarify it.

**Pull request merged**